fixed OverflowException when running as x64

thoemmi opened the pull request:
When running as x64, `ResourceId.IsIntResource` threw an `OverflowException`. (64 bit pointers can't be cast to `uint`)

dblock commented:
Thanks! If you have a repro, a test would be helpful. Otherwise would you be so kind to update CHANGELOG (via `--amend` please) as well?

https://github.com/dblock/resourcelib/blob/master/CHANGELOG.md

thoemmi commented:
I just noticed this issue in my own app and fixed it. I'll try to create a unit test.

I think `--amend` isn't possible as I already published my commit. I'd create another commit which would be added to this PR automatically. Is that ok for you?

dblock commented:
You can always force push, `git push origin <branch> -f`. If you don't mind squashing these please? Just following the protocol :)

dblock commented on CHANGELOG.md:
```diff
@@ -6,6 +6,7 @@
 * Including Vestris.ResourceLib.pdb and Vestris.ResourceLib.xml in release zip for documentation and easier debugging support - [@icnocop](https://github.com/icnocop).
 * [CI](https://ci.appveyor.com/project/dblock/resourcelib): Setup CI on AppVeyor - [@dblock](https://github.com/dblock).
 * [#26](https://github.com/dblock/resourcelib/pull/26): Fixed opening PE files at non-ANSI paths by changing from ANSI to Wide version of PInvoke - [@hypersw](https://github.com/hypersw).
+* [#31](https://github.com/dblock/resourcelib/pull/31): fixed OverflowException when running as x64 - [@thoemmi](https://github.com/thoemmi)
```
This is missing a period at the end, please.

thoemmi replied:
I'm fine with that :laughing: Let me first create a unit test. Afterwards I'll either squash the commits or create a new PR.

dblock replied:
:+1:

thoemmi commented:
I first didn't succeed creating a failing test. But then I switched the target framework of `ResourceLibUnitTests` to **.NET 4.5**. Then most of the existing tests failed too (I also checked .NET 4.0, but that was fine). My original fix and another change in `FontDirectoryResource` made the tests green again :grinning:

Therefore I suggest to change the `ResourceLibUnitTests` to .NET 4.5 permanently. `ResourceLib` can stay at .NET 2.0. What do you think?

(it's already late in Germany, I'm off for today)

thoemmi commented:
BTW, I guess this will fix #22 (and #25) too, because it happens with .NET 4.5.

dblock commented:
@thoemmi Makes total sense, go for it.

thoemmi commented:
I squashed the changes to a single commit:
* Fixed x64 issues in `ResourceId` and `FontDirectoryResource`
* Changed target framework of `ResourceLibUnitTests` to .NET 4.5 to make the x64 issues reproducable
* updated CHANGELOG with fixed typos and punctuation :wink:

dblock commented:
Very well, merged.
